aloha-core/src/test/scala/com/eharmony/aloha/audit/impl/EitherAuditorTest.scala

```diff
@@ -0,0 +1,299 @@
+        assertEquals(NoAggFailMidList, e.failureModelIds)
+        assertEquals(E1, e.errorMsgs)
+        assertEquals(M1, e.missingVarNames)
+    }
+  }
+
+  @Test def testNoAggBothChildSuccess(): Unit = {
+    val v2 = DN.success(IdD, 3d, E2, M2)
+    val v3 = IN.success(IdI, 4, E3, M3)
+    val v1 = SN.failure(IdS, E1, M1, subvalues = Seq(v2, v3))
+
+    v1 match {
+      case Right(_) => fail()
+      case Left(e) =>
+        assertEquals(NoAggFailMidList, e.failureModelIds)
+        assertEquals(E1, e.errorMsgs)
+        assertEquals(M1, e.missingVarNames)
+    }
+  }
+
+  @Test def testNoAggRootSuccess(): Unit = {
+    val v2 = DN.failure(IdD, E2, M2)
+    val v3 = IN.failure(IdI, E3, M3)
+    val v1 = SN.success(IdS, "success", E1, M1, Seq(v2, v3))
+
+    v1 match {
+      case Right(s) => assertEquals("success", s)
+      case Left(_) => fail() // No info
+    }
+  }
+
+  @Test def testNoAgg_fsf(): Unit = {
+    val v3 = IN.failure(IdI, E3, M3)
+    val v2 = DN.success(IdD, 2d, E2, M2, subvalues = Seq(v3))
+    val v1 = SN.failure(IdS, E1, M1, subvalues = Seq(v2))
+
+    v1 match {
+      case Right(_) => fail()
+      case Left(e) =>
+        // Nil because v2 is successful, so it throws away all child info.
+        assertEquals(NoAggFailMidList, e.failureModelIds)
+        assertEquals(E1, e.errorMsgs)
+        assertEquals(M1, e.missingVarNames)
+    }
+  }
+
+  @Test def testNoAgg_ffs(): Unit = {
+    val v3 = IN.success(IdI, 2, E3, M3)
+    val v2 = DN.failure(IdD, E2, M2, subvalues = Seq(v3))
+    val v1 = SN.failure(IdS, E1, M1, subvalues = Seq(v2))
+
+    v1 match {
+      case Right(_) => fail()
+      case Left(e) =>
+        // v3 is not included because it was successful.
+        // None of its children would be included either.
+        assertEquals(NoAggFailMidList, e.failureModelIds)
+        assertEquals(E1, e.errorMsgs)
+        assertEquals(M1, e.missingVarNames)
+    }
+  }
+
+  @Test def testDfsPreOrder(): Unit = {
+    val a = EitherAuditor[Int]
+
+    // Tree:
+    //
+    //   1 +- 2 +- 3
+    //     |    |
+    //     |    +- 4
+    //     |
+    //     +- 5 +- 6
+    //          |
+    //          +- 7
+    //
+    val v3 = a.failure(ModelId(3))
+    val v4 = a.failure(ModelId(4))
+    val v2 = a.failure(ModelId(2), subvalues = Seq(v3, v4))
+
+    val v6 = a.failure(ModelId(6))
+    val v7 = a.failure(ModelId(7))
+    val v5 = a.failure(ModelId(5), subvalues = Seq(v6, v7))
+
+    val v1 = a.failure(ModelId(1), subvalues = Seq(v2, v5))
+
+    val mids = v1.left.get.failureModelIds
+    assertEquals(1 to 7, mids.map(mid => mid.getId()))
+  }
+
+  @Test def testScaladocExample(): Unit = {
+    val a = EitherAuditor[Int]
+
+    // Tree:
+    //
+    //   1:F +-- 2:F +--  3:S
+    //       |       |
+    //       |       +--  4:F
+    //       |
+    //       +-- 5:S +--  6:F
+    //               |
+    //               +--  7:F
+    //
+    val v3 = a.success(ModelId(3), 3, Seq("e3"), Set("m3"))
```

Comment:
Should these be called: `f1`, `f2`, `s3`, `f4`, `s5`, `f6`, `f7`?  Is that any better?